transport_test.go

```diff
@@ -0,0 +1,141 @@
+package lumigotracer
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func TestTransport(t *testing.T) {
+	content := []byte("Hello, world!")
+
+	ctx := context.Background()
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if _, err := w.Write(content); err != nil {
+			t.Fatal(err)
+		}
+	}))
+	defer ts.Close()
+
+	r, err := http.NewRequestWithContext(ctx, http.MethodGet, ts.URL, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testcases := []struct {
+		testname  string
+		transport http.RoundTripper
+	}{
+		{
+			testname:  "http default transport",
+			transport: http.DefaultTransport,
+		},
+		{
+			testname:  "http default transport",
+			transport: nil,
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.testname, func(t *testing.T) {
+			tr := NewTransport(http.DefaultTransport)
+
+			c := http.Client{Transport: tr}
+			res, err := c.Do(r)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			body, err := ioutil.ReadAll(res.Body)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if !bytes.Equal(body, content) {
```

Comment:
Shouldn't we `Equal(expected_body, actual_span_data)` or something similar?

Reply:
same